### PR TITLE
Build new /data page

### PR DESCRIPTION
**Pull request by carkod**
## Done

Build new data page for c.com
## QA

- Check [/data](https://canonical-com-1043.demos.haus/data) against [design](https://www.figma.com/file/17XOOU2kMgAW7CsRhGyu4Y/23.10-C.com---Data-Fabric?type=design&node-id=128-1198&mode=design&t=E7BrAN2kszpcKFSM-0)
- Check against [copy docs](https://docs.google.com/document/d/1r1xnNihaTphO_eikifUEO7ocYfTDnbuRIaIfUCmya1c/edit)
- Click on Get in touch, fill up the form and make sure it works.

- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- [List additional steps to QA the new features or prove the bug has been resolved]

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5872?atlOrigin=eyJpIjoiNGRiMmQzZWRjNDUwNGYyZWFiNmVkMzNmN2Q4ODAzZmQiLCJwIjoiaiJ9

## Screenshots

[if relevant, include a screenshot]

### CODE REVIEW

**Comment by juanruitina** [comment]
- Change title to "Data Fabric | Canonical".
- On mobile, please add some extra spacing between the three cards under "A complete solution for modern data", and also the two cards under "Fully managed or self-managed ‐ it’s your choice", so they are not too close together.
- Drop the colon after "What's included" on the two cards
- Please left align "[Learn about fully-managed ›](http://ubuntu.com/managed/apps)"

**Comment by carkod** [comment]
> * Change title to "Data Fabric | Canonical".
> * On mobile, please add some extra spacing between the three cards under "A complete solution for modern data", and also the two cards under "Fully managed or self-managed ‐ it’s your choice", so they are not too close together.
> * Drop the colon after "What's included" on the two cards
> * Please left align "[Learn about fully-managed ›](http://ubuntu.com/managed/apps)"

All done.

**Comment by lyubomir-popov** [comment]
(not a complete review, will let you know when I'm done):

- double deep spacing here, we need just one:
image

**Comment by juanruitina** [comment]
Thank you, Carlos. A couple more items.

- [ ] As per [the last item in my feedback here](https://github.com/canonical/canonical.com/pull/1046#issuecomment-1733357512) and the copy doc, we should use the original OpenStack logo instead of Canonical's.
- [ ] Please fix the copy doc metatag so it links to [this doc](https://docs.google.com/document/d/1r1xnNihaTphO_eikifUEO7ocYfTDnbuRIaIfUCmya1c/edit?usp=sharing).

**Comment by carkod** [comment]
All done

**Comment by juanruitina** [comment]
We should use the logo with the wordmark here, in line with the copy doc. Other than that, this looks good to me, feel free to change the label to UX+1 once you fix.

**Comment by carkod** [comment]
@lyubomir-popov can I get a review when you can please?

**Comment by lyubomir-popov** [comment]
yes working on it

**Comment by lyubomir-popov** [comment]
image

these headings should be h2s:

image

this should have a line above and be aloigned under the list:
image

these shouldn't be bold:
image

- I would put the "Get in touch" on the same line:
 
image


- broken:
- 
image

**Comment by carkod** [comment]
> image
> 
> these headings should be h2s:
> 
> image
> 
> this should have a line above and be aloigned under the list: image
> 
> these shouldn't be bold: image
> 
> * I would put the "Get in touch" on the same line:
> 
> image
> 
> * broken:
> 
> image

All done.

**Comment by lyubomir-popov** [comment]
please remove the br at small brakpoints (u-hide--small):
image

the label fixed price under the prices should be in normal (regular/400) weight, same as a paragraph. It has now become thin.
I would take it out of the h2 and just use a  paragraph
image

**Comment by carkod** [comment]
> please remove the br at small brakpoints (u-hide--small): image
> 
> the label fixed price under the prices should be in normal (regular/400) weight, same as a paragraph. It has now become thin. I would take it out of the h2 and just use a paragraph image
> 
> grid still broken: image
> 
> this text still isn't aligned with the list: image
> 
> these logos need to start at the left edge on smaller breakpoints: image image
> 
> also broken grid there: image
> 
> removing the medium screen classes fixes it: image

Fixed all the grid issues.

**Comment by lyubomir-popov** [comment]
@carkod thanks for the updates, ca nyou please change one final thing - add a u-no-margin--bottom on the prices as they now have too much space under them:

image

plus 1 already given for a faster merge once the above is fixed

File: templates/data/index.html [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+<section class="p-strip--whitesuru is-shallow u-no-padding--bottom">
+	<div class="row">
+		<div class="col-start-large-4 col-9">
+			<div class="p-section--shallow">
+				<h1>Enterprise Data Fabric solutions for rapid innovation at any scale</h1>
+			</div>
+			<div class="p-section--shallow">
+				<p>Data flows better with the right tools. Canonical’s open source enterprise Data Fabric suite delivers better ROI across clouds.<br />Engineered for the most critical workloads.</p>
+			</div>
+			<hr>
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/index.html:22]: `<hr` tags are not required https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-in-void-element-start-tags-do-not-mark-the-start-tags-as-self-closing

File: templates/data/index.html [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+					<a class="p-button--positive js-invoke-modal" href="/data/contact-us#get-in-touch" aria-controls="data-fabric-modal">Get in touch</a>
+				</p>
+			</div>
+		</div>
+	</div>
+</section>
+
+<section class="p-section">
+	<div class="u-fixed-width p-section">
+		<hr class="p-rule">
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/index.html:34]: As mentioned, not needed `<hr` closing trailling slash tags are not required https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-in-void-element-start-tags-do-not-mark-the-start-tags-as-self-closing

File: templates/data/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (akbarkz) [reply templates/data/index.html:22]: according to HTML coding standards that we have we should selfclose all tags: https://webteam.canonical.com/practices/html-code-standards?_ga=2.80525781.359172970.1695894923-640112026.1695220373#:~:text=Close%20all%20elements%20%2D%20either%20a%20closing%20tag%20(%3Cp%3E...%3C/p%3E)%20or%20self%2Dclosing%20(%3Cimg%20src%3D%22jeff.jpg%22%20/%3E)

[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/index.html:22]: HTML validator gives a warning that closing `<hr />` does not indicate self-closing in the specific case of `<hr />`, for all other tags I do actually close it, in the case of `` and `<input>`

File: templates/data/_data-form.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            <div class="js-formfield">
+              <button class="p-modal__close" aria-controls="data-fabric-modal" aria-label="Close active modal" style="margin-left: -1rem">Close</button>
+              <h3 class="p-heading--5">Fill in this form and we'll be in touch within one working day.</h3>
+              <div class="row u-no-padding">
+                <div class="col-12">
+                  <form action="https://ubuntu.com/marketo/submit" method="post" onsubmit="ga('send', 'Homepage', 'Contact us');" id="mktoForm_{{ formid }}" class="modal-form" id="executive-summit-form">
+                    <ul class="p-list">
+                      <li class="p-list__item">
+                        <label for="Comments_from_lead__c">Tell us about your data related use case:</label>
+                        <textarea required id="Comments_from_lead__c" name="Comments_from_lead__c" rows="4" maxlength="2000"></textarea>
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/_data-form.html]: Addressed

File: templates/data/index.html [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+{% block meta_description %}Canonical Data Fabric Suite enables innovation at scale on any cloud. Advanced open source solutions for data on IaaS and Kubernetes with flexible, enterprise-grade support.{% endblock %}
+
+{% block meta_copydoc %}https://docs.google.com/document/d/1r1xnNihaTphO_eikifUEO7ocYfTDnbuRIaIfUCmya1c/edit{% endblock meta_copydoc %}
+
+{% block body_class %}is-paper{% endblock body_class %}
+
+{% block content %}
+
+<section class="p-strip--whitesuru is-shallow u-no-padding--bottom">
+	<div class="row">
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/index.html]: Not sure if it worked. But I don't see additional space or anything wrong with the indentation?
Screenshot 2023-09-28 at 11 45 28

File: templates/data/_data-form.html [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                        <input required id="email" name="email" maxlength="255" type="email" pattern="^[^ ]+@[^ ]+\.[a-z]{2,26}$" />
+                      </li>
+                      <li class="p-list__item">
+                        <label for="phone">Phone number:</label>
+                        <input id="phone" name="phone" maxlength="255" type="tel" pattern="[0-9\s]+" />
+                            <p class="p-form-help-text" id="exampleHelpTextMessage">Format: only numbers</p>
+                      </li>
+                      <li class="p-list__item">
+                        <label class="p-checkbox">
+                          <input class="p-checkbox__input" value="yes" aria-labelledby="canonicalUpdatesOptIn" name="canonicalUpdatesOptIn" type="checkbox">
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/_data-form.html]: Addressed

File: templates/data/_data-form.html [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                      <li class="p-list__item">
+                        <label for="lastName">Last name*:</label>
+                        <input required id="lastName" name="lastName" maxlength="255" type="text"/>
+                      </li>
+                      <li class="p-list__item">
+                        <label for="company">Company name*:</label>
+                        <input required id="company" name="company" maxlength="255" type="text"/>
+                      </li>
+                      <li class="p-list__item">
+                        <label for="title">Job title *:</label>
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/data/_data-form.html]: Removed space.

File: templates/data/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (akbarkz) [reply templates/data/index.html:22]: ok thanks for investigating it, let's leave it unclosed

**Review by akbarkz — APPROVED** [review]

**Comment by carkod** [comment]
> @carkod I just noticed that the "Get in touch" buttons/links in the hero and at the bottom don't seem to be working, can you check? Marketo ID is 5037 according to the copy doc.

Should be working now.

**Comment by juanruitina** [comment]
Thank you Carlos. To match the [design we are using as reference](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?type=design&node-id=976-3052&mode=design&t=2MrcHgdw2VeuMnBU-0), please:

- [ ] Make the heading an H1 (with H1 styles as well)
- [ ] Remove the HR below the heading
- [ ] Move the line "Fill in this form and we'll be in touch within one working day." below the heading

Also:

- [ ] Make the close "x" button fixed on desktop and tablet (not on mobile), it should always be visible even after the user scrolls
- [ ] Remove the modal margins on mobile, the modal should take over the whole page.

Let me know if you wanna pair over this. All this should be reflected on all the forms.

**Comment by carkod** [comment]
> Thank you Carlos. To match the [design we are using as reference](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?type=design&node-id=976-3052&mode=design&t=2MrcHgdw2VeuMnBU-0), please:
> 
> * [ ]  Make the heading an H1 (with H1 styles as well)
> * [ ]  Remove the HR below the heading
> 
> Also:
> 
> * [ ]  Make the close "x" button fixed on desktop and tablet (not on mobile), it should always be visible even after the user scrolls
> * [ ]  Remove the modal margins on mobile, the modal should take over the whole page.
> 
> Let me know if you wanna pair over this. All this should be reflected on all the forms.

These changes really should be upstreamed to Vanilla...  obviously it takes more time and maybe should't even do custom styles.

**Comment by juanruitina** [comment]
I leave to @lyubomir-popov and you to decide how to handle the more visual bits of my feedback. But this needs to be addressed:

- [ ]  Make the close "x" button fixed on desktop and tablet (not on mobile), it should always be visible even after the user scrolls

**Comment by lyubomir-popov** [comment]
> I leave to @lyubomir-popov and you how to handle the more visual bits of my feedback. But this one needs to be addressed:

I fully agree, thanks for spotting these things I missed.

**Comment by carkod** [comment]
- We've been told multiple times in the past not to have two `<h1` in the same page.
- Shouldn't spacing be consistent across devices/screens? Why are we removing margins only for mobile?

**Comment by juanruitina** [comment]
Same as I shared in https://github.com/canonical/canonical.com/pull/1044: It is acceptable and desirable to adjust margins to device sizes, but I see your point about Vanilla, given the modal is a Vanilla component. I'll make a note to revisit forms in canonical.com, also in terms of accessibility.

I [double checked](https://stackoverflow.com/questions/38448811/is-it-semantically-correct-to-use-h1-in-a-dialog) the heading thing. You are right to use an H2 and style it as an H1. Thank you for flagging it up.

This is OK UX-wise, +1ing it now.

**Comment by carkod** [comment]
I don't think Stackoverflow is this authority in this... but I remember there was an SEO reason not to have multiple `<h1>`, just dev reviews from Scott used to tell me not to do it. I'll ask around

**Comment by anthonydillon** [comment]
> I don't think Stackoverflow is an authority in this... but I remember there was an SEO reason not to have multiple `<h1>`, just dev reviews from Scott used to tell me not to do it. I'll ask around

You should have one top level heading pre document but there are cases with HTML5 where `<article>` resets the document level. Meaning you can have multiple h1 on a page but only one within each article.

**Comment by carkod** [comment]
> > I don't think Stackoverflow is an authority in this... but I remember there was an SEO reason not to have multiple `<h1>`, just dev reviews from Scott used to tell me not to do it. I'll ask around
> 
> You should have one top level heading pre document but there are cases with HTML5 where `<article>` resets the document level. Meaning you can have multiple h1 on a page but only one within each article.

ok, but in this case we don't have `<article>` it's a modal

**Comment by anthonydillon** [comment]
> > > I don't think Stackoverflow is an authority in this... but I remember there was an SEO reason not to have multiple `<h1>`, just dev reviews from Scott used to tell me not to do it. I'll ask around
> > 
> > You should have one top level heading pre document but there are cases with HTML5 where `<article>` resets the document level. Meaning you can have multiple h1 on a page but only one within each article.
> 
> ok, but in this case we don't have `<article>` it's a modal

The dialog element might be classed as a new document. Interesting...

**Pull request merged**